thunder/core/transforms.py

```diff
@@ -1357,6 +1357,28 @@ def _embedding_prim_grad(
 
 register_grad(pids.EMBEDDING, _embedding_prim_grad)
 
+
+def _maximum_grad(a: TensorProxy, b: TensorProxy, /):
+    fwd = prims.maximum(a, b)
+
+    g = get_grad(fwd)
+
+    # Compute sub-gradient if `a == b`
+    a_grad = prims.where(a == b, g / 2, 0.0)
```

Comment:
Nit: the result of `a == b` and `g / 2` could be explicitly saved for reuse (a CSE pass might do it for you, but it's still nice to be explicit when it's readable)